Remove mandatory password cb

julek-wolfssl opened the pull request:
The password callback was being used on every decode attempt. First try to decode without pass then ask for it.

SparkiDev commented on src/ssl.c:
```diff
@@ -5319,13 +5318,22 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
+            if (!usePasswd && format != WOLFSSL_FILETYPE_PEM && info->passwd_cb && algId == 0) {
+                usePasswd = 1;
+                WOLFSSL_MSG("Retrying decode with password callback");
+                goto retry_decode;
```
Avoid goto like this if possible.

dgarske commented:
I am pulling this PR into the OpenVPN PR https://github.com/wolfSSL/wolfssl/pull/2503. Sean, also plan to eliminate the goto. Once I push in 2503 will close this PR.

julek-wolfssl commented:
I removed the goto and moved part of the code to a separate function. I 
hope this solution is better.

On 09/10/2019 00:51, David Garske wrote:
>
> I am pulling this PR into the OpenVPN PR #2503 
> <https://github.com/wolfSSL/wolfssl/pull/2503>. Sean, also plan to 
> eliminate the goto. Once I push in 2503 will close this PR.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/wolfSSL/wolfssl/pull/2505?email_source=notifications&email_token=ALY2OBR4KB2YHR2G4OQRWXLQNUFINA5CNFSM4I6TLKT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAV3VVI#issuecomment-539736789>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ALY2OBTDZR32BR4WQVAZA6TQNUFINANCNFSM4I6TLKTQ>.
>

dgarske commented:
Pulled into #2503 for openvpn port.
